presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -143,6 +152,7 @@ public GlueHiveMetastore(HdfsEnvironment hdfsEnvironment, GlueHiveMetastoreConfi
         this.hdfsContext = new HdfsContext(new Identity(DEFAULT_METASTORE_USER, Optional.empty()));
         this.glueClient = requireNonNull(glueClient, "glueClient is null");
         this.defaultDir = glueConfig.getDefaultWarehouseDir();
+        this.numSegments = Math.min(glueConfig.getMaxGlueConnections(), MAX_SEGMENTS);
```

Comment:
should `glueConfig.getMaxGlueConnections()` be annotated with `@Max(10)`?

thinking aloud -- should number of clients and number of segments be governed by one config?
Especially that now it's not practically possible to disable this functionality

Reply:
I can add another setting to specify the number of segments. Can we default to 5?